pacemaker.spec.in

```diff
@@ -476,7 +476,7 @@ done
 %systemd_preun pacemaker.service
 %else
 /sbin/service pacemaker stop >/dev/null 2>&1 || :
-if [ $1 -eq 0 ]; then
+if [ "$1" -eq 0 ]; then
```

Comment:
-eq is a numeric comparison. According to docs, $1 is 0, 1, or 2 here, so no quoting should be necessary. (If you want to handle a malformed $1, then it should be "$1" = "0" for a string comparison -- technically the quotes around 0 are unnecessary but I find it's visually easier to compare when both sides are consistent.)

Reply:
Or for really an unpredictable shell, `[ "x$1" = x0 ]` but my intention wasn't
to be rabbit-holing here, but rather to provide unified approach plus most
helpful diagnostics should the interface-imposed assumption be broken:

```
$ echo 1; [ -eq 1 ]; echo 2; [ "" -eq 1 ]; echo 3; [ a -eq 1 ]; echo 4; [ = 1 ]; echo 5; [ a = 1 ]
> 1
> bash: [: -eq: unary operator expected
> 2
> bash: [: : integer expression expected
> 3
> bash: [: a: integer expression expected
> 4
> bash: [: =: unary operator expected
> 5
```
It seems that `-eq` with enquoted variable works best.
Quoting numbers is a matter of style, but e.g. for me, vim
highlights numeric and string literals the same, so there's
no gain with extra quotes in this very use case.